sars2seq/translate.py

```diff
@@ -155,62 +92,44 @@ def translateSpike(seq):
     """
     sequence = ''
     current = 0
-    while current + 3 <= len(seq):
+    seqLen = len(seq)
+
+    assert seqLen % 3 == 0, (f'The lenght of a sequence to be translated must '
+                             f'be a multiple of 3 but is {seqLen!r}.')
+
+    while current + 3 <= seqLen:
         codon = seq[current:current+3]
-        if codon in CODONS:
+        if codon in codons:
             # This is a codon that corresponds to a normal amino acid.
-            sequence += CODONS[codon]
+            sequence += codons[codon]
             current += 3
-        elif codon not in CODONS and codon.count('-') == 0:
+        elif codon not in codons and codon.count('-') == 0:
```

Comment:
You do this up to three times in this same block of code. Why not start with `gapCount = codon.count('-')` and use that instead?

Reply:
Done.